ui/src/QrScan.tsx

```diff
@@ -0,0 +1,259 @@
+      },
+      [theme.breakpoints.up('md')]: {
+        maxHeight: '550px',
+        maxWidth: '650px',
+        margin: '2rem',
+      },
+    },
+    gridContainerMultiple: {
+      display: 'flex',
+      flexDirection: 'row',
+      flexGrow: 1,
+      alignItems: 'right',
+      justifyContent: 'right',
+      paddingBottom: '1rem',
+    },
+    gridItem: {
+      display: 'flex',
+      position: 'relative',
+      alignItems: 'center',
+      justifyContent: 'center',
+      width: '100%',
+      height: '100%',
+    },
+  };
+
+  const StyledImg = styled('img')({
+    position: 'relative',
+    [theme.breakpoints.down('md')]: {
+      maxHeight: '550px',
+      maxWidth: '300px',
+    },
+    [theme.breakpoints.up('md')]: {
+      maxHeight: '550px',
+      maxWidth: '650px',
+    },
+    objectFit: 'contain',
+    zIndex: '2',
+  });
+
+  const confirmSHLCreation = () => {
+    if (window.confirm('SMART Health Card successfully scanned. Create new SMART Health Link?') === true) {
+      return true;
+    } else {
+      return false;
+    }
+  };
+
+  const handleError = useCallback(() => {
+    navigate('/error');
+  }, [navigate]);
+
+  useEffect(() => () => {
+    if (runningQrScanner.current) {
+      runningQrScanner.current.stop();
+    }
+  }, [])
+
+
+  const videoRef = useRef<any>(null);
+  // Get user media when the page first renders, and feed into createQrScanner()
+  useEffect(() => {
+    const getUserMedia = async () => {
+      try {
+        if (videoRef.current) {
+          const stream = await navigator.mediaDevices.getUserMedia({
+            audio: false,
+            video: { facingMode: 'environment' },
+          });
+          videoRef.current.srcObject = stream;
+        }
+      } catch (err: any) {
+        throw new Error(`Cannot access video: ${err.message}.`);
+      }
+    }; 
+
+    /**
+    * Create QrScanner instance using video element and specify result/error conditions
+    * @param {HTMLVideoElement} videoElement HTML video element
+    */
+    const createQrScanner = async (videoElement: any) => {
+      if (!videoElement) {
+        if (runningQrScanner.current) {
+          qrScan.destroy();
+        }
+        return;
+      }
+      qrScan = new QrScanner(
+        videoElement,
+        (results) => {
+          setScannedData(results.data);
+        },
+        {
+          preferredCamera: 'environment',
+          calculateScanRegion: (video) => ({
+            // define scan region for QrScanner
+            x: 0,
+            y: 0,
+            width: video.videoWidth,
+            height: video.videoHeight,
+          }),
+        },
+      );
+      runningQrScanner.current = qrScan;
+
+      qrScan.start();
+    };
+
+    if (!runningQrScanner.current) {
+      getUserMedia().then(async () => {
+        await createQrScanner(videoRef.current);
+      }, handleErrorFallback);
+    }
+
+    return () => {
+      if (runningQrScanner.current) runningQrScanner.current.stop();
```

Comment:
There's a standalone effect above that also tries to stop the scanner once, on mount. Can you help me understand when/why the scanner would be running at mount time if this cleanup code is in place?

Reply:
I removed the standalone effect from above - that was leftover from code used in the SHC web verifier app to switch the camera stream for mobile devices. In certain cases, the qrscanner's video stream would get interrupted and throw a DOMException. That extra handling is not needed here since the camera stream does not change when scanning